app/android.pri

```diff
@@ -1,49 +1,74 @@
 android {
     message("Building ANDROID")
-    message("ANDROID Platform: $${ANDROID_TARGET_ARCH}")
+    message("ANDROID Platform: $${QT_ARCH}")
 
     DEFINES += MOBILE_OS
 
-    isEmpty(QGIS_INSTALL_PATH) {
-      error("Missing QGIS_INSTALL_PATH")
+    isEmpty(INPUT_SDK_PATH) {
+      error("Missing INPUT_SDK_PATH")
     }
 
-    # using installed QGIS
-    QGIS_PREFIX_PATH = $${QGIS_INSTALL_PATH}
-    QGIS_LIB_DIR = $${QGIS_INSTALL_PATH}/lib
-    QGIS_INCLUDE_DIR = $${QGIS_INSTALL_PATH}/include/qgis
+    INPUT_SDK_ARCH_PATH = $${INPUT_SDK_PATH}/$${QT_ARCH}
+    INPUT_SDK_LIB_PATH = $${INPUT_SDK_ARCH_PATH}/lib
+    INPUT_SDK_INCLUDE_PATH = $${INPUT_SDK_ARCH_PATH}/include
 
-    exists($${QGIS_LIB_DIR}/libqgis_core_$${ANDROID_TARGET_ARCH}.so) {
-      message("Building from QGIS: $${QGIS_LIB_DIR}/libqgis_core_$${ANDROID_TARGET_ARCH}.so")
+    exists($${INPUT_SDK_LIB_PATH}/libqgis_core.a) {
+      message("Building from QGIS: $${INPUT_SDK_LIB_PATH}/libqgis_core.a")
     } else {
-      error("Missing QGIS Core library in $${QGIS_LIB_DIR}/libqgis_core_$${ANDROID_TARGET_ARCH}.so")
+      error("Missing QGIS Core library in $${INPUT_SDK_LIB_PATH}/libqgis_core.a")
     }
+    
+    INCLUDEPATH += $${INPUT_SDK_INCLUDE_PATH}
+    INCLUDEPATH += $${INPUT_SDK_INCLUDE_PATH}/qgis
+    
+    # by default QMake eats linked libs if they are mentioned multiple times
+    # https://stackoverflow.com/questions/18327959/qmake-how-to-link-a-library-twice/18328971
+    CONFIG += no_lflags_merge
 
-    INCLUDEPATH += $${QGIS_INCLUDE_DIR}
-    LIBS += -L$${QGIS_LIB_DIR}
-    LIBS += -lqgis_core_$${ANDROID_TARGET_ARCH}
+    LIBS += -lc++_shared
```

Comment:
is this needed at the end? (remember it was part of our testing)

Reply:
it is not